modules/IngestDrugSynonymsSync.py

```diff
@@ -0,0 +1,228 @@
+import requests
+import pandas as pd
+import re
+import json
+import sys
+from pathlib import Path
+import xlrd
+import csv
+import os
+import tempfile
+import numpy as np
+from typing import Optional
+from functools import partial
+import logging
+logger = logging.getLogger('ds')
+
+
+class IngestDrugSynonyms():
+
+    def __init__(self, configPath: Path = Path("config.json")):
+        self.check_config(configPath)
+        self.url_international: str = os.environ["URL_INT"] if isinstance(
+            os.environ["URL_INT"], str) else None
+        self.url_USA: str = os.environ["URL_USA"] if isinstance(
+            os.environ["URL_USA"], str) else None
+        self.url_drugbank: str = os.environ["URL_DRUGBANK"] if isinstance(
+            os.environ["URL_DRUGBANK"], str) else None
+        self.query_keywords: [] = os.environ["QUERY_KEYWORDS"].split(
+            ",") if isinstance(os.environ["QUERY_KEYWORDS"], str) else None
+
+    @staticmethod
+    def check_config(configPath: Path = Path("config.json")):
+        def load_config(configPath: Path):
+            with open(configPath) as f:
+                config = json.load(f)
+                for key in config:
+                    os.environ[key] = config[key]
+        rel_path: Path = Path(os.path.realpath(
+            __file__)).parents[0] / configPath
+        if rel_path.exists:
+            load_config(rel_path)
+        elif configPath.exists:
+            load_config(configPath)
+        else:
+            logger.warning(
+                "Could not load config file from: {}".format(configPath))
+
+    @staticmethod
+    def api(query, from_study, to_study, url):
+        url = url.format(query, from_study, to_study)
+        response = requests.request("GET", url)
+        return response.json()
+
+    def api_wrapper(self, query, from_study):
+        return self.api(query, from_study, from_study+99, self.url_USA)
+
+    def getAllStudiesByQuery(self, query: str) -> list:
+        logger.info("> STARTING scraping with '{}' keyword".format(query))
+        studies: list = []
+        from_study = 1
+        temp = self.api_wrapper(query, from_study)
+        nstudies = temp['FullStudiesResponse']['NStudiesFound']
+        logger.info(
+            "> {} studies found by '{}' keyword".format(nstudies, query))
+        if nstudies > 0:
+            studies = temp['FullStudiesResponse']['FullStudies']
+            for study_index in range(from_study+100, nstudies, 100):
+                temp = self.api_wrapper(query, study_index)
+                studies.extend(temp['FullStudiesResponse']['FullStudies'])
+
+        return studies
+
+    @staticmethod
+    def xls_handler(r):
```

Comment:
AFAICT, this downloads an xls, rewrites into csv, and rereads the csv w/ pandas, and returns the result

Pandas has an xls reader, and may be able to work directly on the bytes buffer. A bit surprising to see like this, and I recall discussion of some formatting issues encountered along the way.

Reply:
(not urgent if works, to be clear)